File: Iv Layhort/Scala-Layhort/src/Task_7/Exercise_1_Kangaroo.scala
package Task_7
import scala.io.StdIn._
import util.control.Breaks._
object Exercise_1_Kangaroo {
  def main(args: Array[String]): Unit = {
    println("Enter the first kangaroo starting point :")
    var firstKangarooStartingPoint = readInt()
    println("Enter the first kangaroo jumping steps :")
    var firstKangarooJumpingSteps = readInt()
    println("Enter the second kangaroo starting point :")
    var secondKangarooStartingPoint = readInt()
    println("Enter the second kangaroo jumping steps :")
    var secondKangarooJumpingSteps = readInt()
    breakable{
      while(firstKangarooStartingPoint<=secondKangarooStartingPoint){
        if(firstKangarooStartingPoint == secondKangarooStartingPoint){
          println("Yes")
          break()
        }
        firstKangarooStartingPoint += firstKangarooJumpingSteps
        secondKangarooStartingPoint+= secondKangarooJumpingSteps
      }
      println("No")
    }
  }
}

File: Iv Layhort/Java_Layhort/src/Task_7/Exercise_1_Kangaroo.java
package Task_7;

import java.util.*;
public class Exercise_1_Kangaroo {
    public static void main(String[] args) {
        Scanner input  = new Scanner(System.in);
        int firstKangarooStartingPoint, secondKangarooStartingPoint,firstKangarooJumpingSteps,secondKangarooJumpingSteps;
        System.out.println("Enter the first kangaroo starting point :");
        firstKangarooStartingPoint = input.nextInt();
        System.out.println("Enter the first kangaroo jumping steps :");
        firstKangarooJumpingSteps = input.nextInt();
        System.out.println("Enter the second kangaroo starting point :");
        secondKangarooStartingPoint = input.nextInt();
        System.out.println("Enter the second kangaroo jumping steps :");
        secondKangarooJumpingSteps = input.nextInt();

        while (firstKangarooStartingPoint<=secondKangarooStartingPoint){
            if(firstKangarooStartingPoint == secondKangarooStartingPoint){
                System.out.println("Yes");
                   System.exit(0);
            }
                firstKangarooStartingPoint+=firstKangarooJumpingSteps;
                secondKangarooStartingPoint+=secondKangarooJumpingSteps;
        }
        System.out.println("No");
    }
}
